WindowsDevicePortalWrapper/WindowsDevicePortalWrapper.Shared/HoloLens/PerceptionSimulationPlayback.cs

```diff
@@ -49,15 +55,183 @@ public partial class DevicePortal
+                payload);
+
+            using (Stream dataStream = await this.Get(uri))
+            {
+                if ((dataStream != null) &&
+                    (dataStream.Length != 0))
+                {
+                    // Try to get the session state.
+                    try
+                    {
+                        DataContractJsonSerializer serializer = new DataContractJsonSerializer(typeof(HolographicSimulationPlaybackSessionState));
+                        HolographicSimulationPlaybackSessionState sessionState = (HolographicSimulationPlaybackSessionState)serializer.ReadObject(dataStream);
+                        playbackState = sessionState.State;
+                    }
+                    catch 
+                    {
+                        // We did not receive the session state, check to see if we received a simulation error.
+                        dataStream.Position = 0;
+                        DataContractJsonSerializer serializer = new DataContractJsonSerializer(typeof(HolographicSimulationError));
+                        HolographicSimulationError error = (HolographicSimulationError)serializer.ReadObject(dataStream);
+                        throw new InvalidOperationException(error.Reason);
+                    }
+                }
+            }
+
+            return playbackState;
+        }
+
+        /// <summary>
+        /// Loads the specified Holographic Simulation recording.
+        /// </summary>
+        /// <param name="name">The name of the recording to load (ex: testsession.xef).</param>
+        /// <remarks>This method is only supported on HoloLens devices.</remarks>
+        public async Task LoadHolographicSimulationRecording(string name)
+        {
+            if (!Utilities.IsHoloLens(this.Platform, this.DeviceFamily))
+            {
+                throw new NotSupportedException("This method is only supported on HoloLens.");
+            }
+
+            string payload = string.Format(
+                "recording={0}",
+                name);
+
+            await this.Post(HolographicSimulationPlaybackSessionFileApi, payload);
+        }
+
+        /// <summary>
+        /// Unloads the specified Holographic Simulation recording.
+        /// </summary>
+        /// <param name="name">The name of the recording to unload (ex: testsession.xef).</param>
+        /// <remarks>This method is only supported on HoloLens devices.</remarks>
+        public async Task UnloadHolographicSimulationRecording(string name)
+        {
+            if (!Utilities.IsHoloLens(this.Platform, this.DeviceFamily))
+            {
+                throw new NotSupportedException("This method is only supported on HoloLens.");
+            }
+
+            string payload = string.Format(
+                "recording={0}",
+                name);
+
+            await this.Delete(HolographicSimulationPlaybackSessionFileApi, payload);
+        }
 
         #region Data contract
+        /// <summary>
```

Comment:
I expect StyleCop is also going to complain that this enum isn't at the top of the file.